samcli/commands/validate/validate.py

```diff
@@ -15,13 +17,25 @@
 from samcli.lib.telemetry.metric import track_command
 from samcli.cli.cli_config_file import configuration_option, TomlProvider
 from samcli.lib.utils.version_checker import check_newer_version
+from samcli.commands._utils.click_mutex import ClickMutex
 
+LOGGER = logging.getLogger("cfnlint")
+LOGGER.propagate = False
 
 @click.command("validate", short_help="Validate an AWS SAM template.")
 @configuration_option(provider=TomlProvider(section="parameters"))
 @template_option_without_build
 @aws_creds_options
 @cli_framework_options
+@click.option(
+    "--lint",
+    is_flag=True,
+    is_eager=True,
+    help="Run linting validation on template through cfn-lint. "
+     "For more information, see: https://github.com/aws-cloudformation/cfn-lint",
+    cls=ClickMutex,
+    incompatible_params=["config-env", "config-file", "profile", "region"]
```

Comment:
why are these incompatible? you could very well have the template file be passed in through a configuration parameter within the 'validate' space such that it is `default.validate.parameters`, which has a template specified.

I see that cfn-lint has a way to pass in regions to its command line. what does that mapping mean here?

Does cfn-lint have any environment variables that it obeys? can they be used in conjunction through `sam validate --lint` ?

Reply:
Hmmm, I see fair point - missed that. Would setting the template in the config-file also pass the template file path to the do_cli method? As for regions, I can include that as an option in the _lint method as well.